### PR TITLE
Add possibility of random drawing with replacement

### PR DESCRIPTION
**Pull request by terezakroupa**
- new method in DataSetGenerator
- add unit tests for drawing with and without replacement

### CODE REVIEW

File: test/unit/DrawNoReplTest.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#include <catch.hpp>
```
Comment (jackdunger) [inline test/unit/DrawNoReplTest.cpp]: Do you mind combining this and the other test file into a single file named after the class you're testing i.e. two test cases in `DataSetGeneratorTest.cpp`

File: src/data/DataSetGenerator.cpp [hunk 2]
```diff
@@ -163,6 +165,49 @@ DataSetGenerator::RandomDrawsNoReplacement(size_t handleIndex_, int nEvents_,
   return;
 }
 
+
+void
+DataSetGenerator::RandomDrawsWithReplacement(size_t handleIndex_, int nEvents_, 
+					          OXSXDataSet& outData_
+					     ){
+
+  std::vector<size_t>& eventIndices = fEventIndicies[handleIndex_];
```
Comment (jackdunger) [inline src/data/DataSetGenerator.cpp]: do you need `eventIndices` at all in this method?

Comment (terezakroupa) [inline src/data/DataSetGenerator.cpp]: hmm.. maybe in the case when fMaxs is empty?

File: test/unit/DrawNoReplTest.cpp (hunk 1 above)
Comment (terezakroupa) [inline test/unit/DrawNoReplTest.cpp]: Would making one test case in one file with twice as many sections work as well? That would save creating the original datasets twice.

**Comment by jackdunger** [comment]
Hi @terezakroupa  are you finished with this?

**Comment by terezakroupa** [comment]
Hi, yes, sorry for the late reaction. It should be ready. The two tests are combined and I left the method as it was initially after what we discussed in person.

**Comment by jackdunger** [comment]
Excellent, thanks for this

**Pull request merged**